src/Streamly/Coreutils/Ln.hs

```diff
@@ -0,0 +1,51 @@
+-- |
+-- Module      : Streamly.Coreutils.Ln
+-- Copyright   : (c) 2022 Composewell Technologies
+-- License     : Apache-2.0
+-- Maintainer  : streamly@composewell.com
+-- Stability   : experimental
+-- Portability : GHC
+--
+-- create a link to TARGET with the given name
+
+module Streamly.Coreutils.Ln
+    ( ln
+    , setForce
+    , setSymbolic
+    )
+where
+
+import Data.Default.Class (Default(..))
+import qualified System.Posix.Files as Posix
+import Streamly.Coreutils.Common (Switch(..))
+
+data Ln = Ln
+    { force :: Switch
+    , symbolic :: Switch
+    }
+
+instance Default Ln where
+    def = Ln Off Off
+
+setForce :: Switch -> Ln -> Ln
+setForce opt ln0 = ln0 {force = opt}
+
+setSymbolic :: Switch -> Ln -> Ln
+setSymbolic opt ln0 = ln0 {symbolic = opt}
+
+ln :: Ln -> FilePath -> FilePath -> IO ()
```

Comment:
Let's make the signature:

```
ln :: (Ln -> Ln) -> FilePath -> FilePath -> IO ()
```

You would have to change your code below to something like this:

```
ln f src tgt = 
    let opt = f def
     in ...
```

For default command we will use `ln id`.

With this we also won't need a default instance for configuration. Instead of `def` above you could just use a `defaultConfig :: Ln` which you can define as default configuration record in this module.

This is not new we are using exactly the same method in streamly FileSystem.Event module.